src/__experimental__/components/decimal/decimal.spec.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import TestUtils from 'react-dom/test-utils';
+import Decimal from './decimal.component';
+import Textbox from '../textbox/textbox.component';
+
+function render(props) {
+  const onChange = props.onChange ? props.onChange : () => true;
+
+  return shallow(
+    <Decimal
+      onChange={ onChange }
+      value={ props.value }
+      precision={ props.precision }
+    />
+  );
+}
+
+function renderIntoDocument(props) {
+  return TestUtils.renderIntoDocument(
+    <Decimal value={ props.value } precision={ props.precision } />
+  );
+}
+
+describe('Decimal', () => {
+  describe('Input validation', () => {
+    it('renders the correct value', () => {
+      const input = render({ value: '9.87' });
+      expect(input.find(Textbox).prop('value')).toEqual('9.87');
+    });
+
+    it('renders a value from props exactly even when invalid', () => {
+      const input = render({ value: '12abc.85' });
+      expect(input.find(Textbox).prop('value')).toEqual('12abc.85');
+    });
+
+    it('updates an external state onChange', () => {
+      const onChange = jest.fn();
+      const input = render({ onChange });
+      input.instance().onChange({ target: { value: '14.79' } });
+      expect(onChange).toHaveBeenCalledWith({ target: { value: '14.79' } });
+    });
+
+    it('does not allow the user to enter letters or special characters', () => {
+      const input = render({ value: '12.34' });
+      input.instance().onChange({ target: { value: '1hello$1.27' } });
+      expect(input.find(Textbox).prop('value')).toEqual('12.34');
+    });
+
+    it('does not allow the user to enter commas after the decimal point', () => {
+      const input = render({ value: '34.56' });
+      input.instance().onChange({ target: { value: '34.5,,,,6' } });
+      expect(input.find(Textbox).prop('value')).toEqual('34.56');
+    });
+
+    it('forces change event if user blurs', () => {
+      const onChange = jest.fn();
+      const input = render({ onChange });
+      input.instance().onBlur({ target: { value: '1' } });
+      expect(onChange).toHaveBeenCalled();
+    });
+
+    it('formats with delimiters with input is not active', () => {
+      const instance = renderIntoDocument({ value: '1234567.00' });
+      expect(instance.formatValue()).toEqual('1,234,567.00');
```

Comment:
You repeat `expect(instance.formatValue()).toEqual(/* */)` a lot here. You could put it in a helper function. 

Also, it may not be completely clear what the instance method `formatValue` actually means in the context of the component.

Reply:
Added in a couple of helper functions to remove duplicated `expect`s.

I think it's fairly clear what `formatValue` is doing, happy to take suggestions though?

Reply:
It's more that this way of testing exposes implementation detail. Our assertions shouldn't rely on the specific names of methods, but rather their results.